src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -3,30 +3,58 @@
 
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Linq;
+using System.Reflection;
 using System.Text.Json.Serialization;
-using System.Text.Json.SourceGeneration.Reflection;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Text;
 
 namespace System.Text.Json.SourceGeneration
 {
     public sealed partial class JsonSourceGenerator
     {
-        private sealed class Emitter
+        private sealed partial class Emitter
         {
+            // Literals in generated source
             private const string RuntimeCustomConverterFetchingMethodName = "GetRuntimeProvidedCustomConverter";
-
-            private const string JsonContextDeclarationSource = "internal partial class JsonContext : JsonSerializerContext";
-
             private const string OptionsInstanceVariableName = "Options";
-
-            private const string PropInitFuncVarName = "PropInitFunc";
-
-            private const string JsonMetadataServicesClassName = "JsonMetadataServices";
-
+            private const string PropInitMethodNameSuffix = "PropInit";
+            private const string SerializeMethodNameSuffix = "Serialize";
             private const string CreateValueInfoMethodName = "CreateValueInfo";
-
+            private const string DefaultOptionsStaticVarName = "s_defaultOptions";
+            private const string DefaultContextBackingStaticVarName = "s_defaultContext";
+            private const string WriterVarName = "writer";
+            private const string ValueVarName = "value";
+
+            private static AssemblyName _assemblyName = typeof(Emitter).Assembly.GetName();
+            private static readonly string s_generatedCodeAttributeSource = $@"
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute(""{_assemblyName.Name}"", ""{_assemblyName.Version}"")]";
+
+            // global::fully.qualified.name for referenced types
+            private const string ActionTypeRef = "global::System.Action";
```

Comment:
I don't see this being used. Are all of these const strings used?

Reply:
It wasn't used. I've removed it and verified that others are used.